yt_dlp/extractor/omnyfm.py

```diff
@@ -0,0 +1,70 @@
+import functools
+import json
+import math
+
+from .common import InfoExtractor
+from ..utils import (
+    InAdvancePagedList,
+    clean_html,
+    float_or_none,
+    get_element_by_id,
+    int_or_none,
+    str_or_none,
+    traverse_obj,
+    unified_strdate,
+    url_or_none,
+)
+
+
+class OmnyFMShowIE(InfoExtractor):
+    IE_NAME = 'omnyfm:show'
+    _VALID_URL = r'https?://omny\.fm/shows/(?P<id>[\w-]+)'
+    _EMBED_REGEX = [r'<iframe[^>]+?src=(?:["\'])(?P<url>https?://omny\.fm/shows/.+?)\1']
+    _PAGE_SIZE = 10
+    _TESTS = [{
+        'url': 'https://omny.fm/shows/league-leaders',
+        'info_dict': {
+            'id': 'bbe146d4-9bee-4763-b785-ad830009a23f',
+            'title': 'League Leaders with Nicole Livingstone',
+        },
+        'playlist_mincount': 15,
+    }, {
+        'url': 'https://omny.fm/shows/afl-daily',
+        'only_matching': True,
+    }]
+
+    def _fetch_page(self, org_id, playlist_id, page):
+        return self._download_json(f'https://api.omny.fm/orgs/{org_id}/programs/{playlist_id}/clips?cursor={page}&pageSize={self._PAGE_SIZE}', f'{playlist_id}_{page}')
+
+    def _entries(self, org_id, playlist_id, first_page_data, page):
+        data = first_page_data if not page else self._fetch_page(org_id, playlist_id, page + 1)
+        for clip in data.get('Clips', {}):
+            yield traverse_obj(clip, {
+                'id': ('Id', {str_or_none}),
+                'title': ('Title', {str_or_none}),
+                'description': ('Description', {clean_html}),
+                'thumbnail': (('ImageUrl', 'ArtworkUrl'), {url_or_none}, any),
+                'duration': ('DurationSeconds', {float_or_none}),
+                'url': ('AudioUrl', {url_or_none}),
+                'season_number': ('Season', {int_or_none}),
+                'episode_number': ('Episode', {int_or_none}),
+                'timestamp': ('PublishedUtc', {unified_strdate}, {int_or_none}),
+                'filesize': ('PublishedAudioSizeInBytes', {int}),
+            })
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        page_url = 'https://omny.fm/shows/' + display_id
```

Comment:
Why not use the URL directly?

Reply:
There are multiple layouts available for this site where the data is present.
So, I thought its better to handle it in such a way that multiple variations of the urls are supported.
example: all these variations are the same podcast but with different endpoints and page layout.
[variation-1](https://omny.fm/shows/tagged/playlists/podcast/embed?style=cover&image=1&share=1&download=1&description=1&follow=1&playlistimages=1&playlistshare=1&foreground=6d4d8f&background=f5f5f5&highlight=feacd2)
[variation-2](https://omny.fm/shows/tagged)
[variation-3](https://omny.fm/shows/tagged/playlists)
[variation-4](https://omny.fm/shows/tagged/playlists/podcast.rss)

Please let me know if this can be handled in a better way.

Reply:
Hmm, you just need the substring after the "shows/"? Let's update `_VALID_URL` :

```diff
-_VALID_URL = r'https?://omny\.fm/shows/(?P<id>[\w-]+)'
+_VALID_URL = r'https?://omny\.fm/shows/(?P<id>[^/]+)'

```

<br>

PS: since we are talking about URLs, directly putting those URLs in the comment is OK and would make things a bit more straightforward.

Reply:
Understood. Will keep this in mind. I've updated `_VALID_URL`. Thanks